privacyidea/lib/policy.py

```diff
@@ -2386,4 +2409,22 @@ def admin_or_user(cls, g, action, realm):
         return cls(g, name=None, scope=scope, realm=realm, active=True,
                    resolver=None, user=username, user_object=None,
                    client=g.client_ip, action=action, adminrealm=adminrealm, time=None,
-                   sort_by_priority=True)
\ No newline at end of file
+                   sort_by_priority=True)
+
+    @classmethod
+    def generic(cls, g, scope=None, realm=None, resolver=None, user=None, user_object=None,
+                client=None, action=None, adminrealm=None, adminuser=None, time=None,
```

Comment:
we can determine the `client` parameter from the `g` object so we would only need to pass the client if it differs from `g`.

Reply:
done in 35f0963b